Sources/TuistAutomation/Utilities/Formatter.swift

```diff
@@ -14,7 +14,6 @@ final class Formatter: Formatting {
     }
 
     func buildArguments() throws -> [String] {
-        let xcbeautifyPath = try binaryLocator.xcbeautifyPath()
-        return [xcbeautifyPath.pathString]
+        return [try binaryLocator.xcbeautifyCommand()]
     }
```

Comment:
How do I test that this continues to work?

Reply:
If I try to run `swift run tuist test` I got:

```
We received an error that we couldn't handle:
    - Localized description: The file “vendor xcbeautify” doesn’t exist.
    - Error: Error Domain=NSCocoaErrorDomain Code=4 "The file “vendor xcbeautify” doesn’t exist." UserInfo={NSFilePath=/Users/df/Developer/tuist/tuist/swift run --package-path /Users/df/Developer/tuist/tuist/projects/tuist/vendor xcbeautify}

If you think it's a legit issue, please file an issue including the reproducible steps: https://github.com/tuist/tuist/issues/new/choose
Consider creating an issue using the following link: https://github.com/tuist/tuist/issues/new/choose
```

I see the command is executed as `return System.shared.publisher(command, pipeTo: try! formatter.buildArguments())`

Might it depend on our implementation of `System.publisher`?

Reply:
> The file “vendor xcbeautify” doesn’t exist.

"vendor xcbeautify" looks a bit suspicious. For the implementation, is the test going through the `#if DEBUG` part of the branch or the other? Based on the error I'd say is the `#if DEBUG`. In that case, I'd check what command we are running with LLDB:

```
swift run --package-path \(bundlePath) xcbeautify
```

I suggest running that command from your terminal. If it succeeds, I'd then consider turning that from `String` into `[String]` where the first element represents the command to execute and the rest of the arguments being passed. I'd also run it through `xcrun` to ensure it uses the Swift associated with the selected Xcode version. Otherwise, we might get inconsistent results in environments where there's a version of Swift configured besides the one that's configured by Xcode. 

Let me know if that helps.

Reply:
Yes the command is printed in the error message and works fine from shell, let me try what you've suggested

Reply:
mmm, actually it looks like it's trying to run the `swift` binary in tuist folder:
```
    - Error: Error Domain=NSCocoaErrorDomain Code=4 "The file “vendor xcbeautify” doesn’t exist." UserInfo={NSFilePath=/Users/df/Developer/tuist/tuist/swift run --package-path /Users/df/Developer/tuist/tuist/projects/tuist/vendor xcbeautify}
```